### DISCUSSION TITLE
esp32-s3-4848s040 Backlight works inverted. or not at all

### DISCUSSION THREAD

**Discussion by JL-Productions** [HASwitchPlate/openHASP#723] [Hardware]
Backlight is inverted. yes i configured in the web interface as inverted. because otherwise i dont see anything .

If backlight is inverted in web interface
  I can control the backlight via the button in homeassistant
  I see something when i turn off my backlight in home assistant. but dont when i turn it on in home assistant. (which looks normal 
         because i inverted the display
  When display is turned off in home assistant so i see my plate and i touch the display to control a button the display turns off. and if i 
         press again on the same place it controls a device. but not on initial press. 
  
If backlight is not inverted in web interface
  I cannot control backlight at all in home assistant.
  I dont see anything from the moment it connects receive an ip address and connects to home assistant. 
  I can control devices when i press at the blackscreen if i press on a location a button is located.

**Comment by jl-p** [comment]
I found that the screen was not on 100% brightness. but 99%. and then its black

**Comment by fvanroie** [comment]
It seems to me that the Guiton 4" doesn't support backlight dimming. Check out [this issue](https://github.com/HASwitchPlate/openHASP/issues/603) for more information.
If you are able to compile custom firmware, you can experiment with different settings for `BACKLIGHT_FREQUENCY`.

**Reply by FreeBear-nc** [reply]
Got a Guiton 4" in front of me - It does support backlight dimming. Not tried controlling it from HA though.
What it doesn't have is mood lighting.

**Comment by smcgann99** [comment]
I tested this screen a while ago and found 100Hz works quite well. I thought this was included in the firmware now ?
 https://discord.com/channels/538814618106331137/692393568266092617/1220529393538044025

**Reply by fvanroie** [reply]
Yes, it was changed [2 months ago](https://github.com/HASwitchPlate/openHASP/blob/82e4095dd4626465478914339a511de93aeaf172/user_setups/esp32s3/esp32-s3-4848S040.ini#L55).

It can be dependent on devices/model/revision/PCB/country grid frequency/...
Without knowing more details, like full firmware version, it is hard to tell what's the issue.
Does backlight dimming work in other firmware?
